AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/InitializeMutation.dfy

```diff
@@ -195,14 +196,24 @@ module {:options "/functionSyntax:4" } InternalInitializeMutation {
       Types.KeyStoreAdminException(
         message := "Active Branch Key Item read from storage is malformed!")
     );
-
-      // TODO-HV-2-M2: Support items in HV-2
+    // TODO-HV-2-M2: Support items in HV-2
     :- Need(
       readItems.ActiveItem.EncryptionContext[Structure.HIERARCHY_VERSION] == Structure.HIERARCHY_VERSION_VALUE_1,
       Types.KeyStoreAdminException(
         message := "At this time, Mutations ONLY support HV-1; BK's Active Item is HV-2.")
     );
-
+    if (
+        && input.Mutations.TerminalHierarchyVersion.Some?
+        && input.Mutations.TerminalHierarchyVersion.value.v2?
+      ) {
+      :- Need(
+        HvUtils.HasUniqueTransformedKeys?(readItems.ActiveItem.EncryptionContext),
+        Types.UnexpectedStateException(
+          message :=
+            KeyStoreErrorMessages.NOT_UNIQUE_BRANCH_KEY_CONTEXT_KEYS
+        )
+      );
```

Comment:
This is good and is a start, 
but I believe we will need to also check
the union of the terminalEC and the inferredEC.

See lines 252 to 278;
I am still trying to imagine an edge case b/w new EC and old EC...